File: core/src/main/java/com/lomekwi/cave/pipeline/audio/AudFrame.java
package com.lomekwi.cave.pipeline.audio;

import com.lomekwi.cave.pipeline.Frame;
import com.lomekwi.cave.timeline.Track;

public class AudFrame extends Frame {
    private float[] samples;
    private final int sampleRate;
    private final int channels;
    private long time;

    public AudFrame(int sampleRate, int channels, Track track) {
        super(track);
        this.sampleRate = sampleRate;
        this.channels = channels;
    }

    public float[] getSamples() {
        return samples;
    }

    public AudFrame setSamples(float[] samples) {
        this.samples = samples;
        return this;
    }

    public int getSampleRate() {
        return sampleRate;
    }

    public long getTime() {
        return time;
    }

    public AudFrame setTime(long time) {
        this.time = time;
        return this;
    }

    @Override
    public void close() {
        samples = null;
    }
}

File: core/src/main/scala/com/lomekwi/cave/pipeline/audio/AudFrame.scala
package com.lomekwi.cave.pipeline.audio

import com.lomekwi.cave.pipeline.Frame
import com.lomekwi.cave.pipeline.Source
import scala.compiletime.uninitialized

class AudFrame(sampleRate0: Int, trackIndex: Int, source: Source[?]) extends Frame(trackIndex, source) {
  private var samples: Array[Float] = uninitialized
  private final val sampleRate: Int = sampleRate0
  private var time: Long = 0

  def this(sampleRate: Int, trackIndex: Int) = {
    this(sampleRate, trackIndex, null)
  }

  def getSamples: Array[Float] = {
    samples
  }

  def setSamples(samples: Array[Float]): AudFrame = {
    this.samples = samples
    this
  }

  def getSampleRate: Int = {
    sampleRate
  }

  def getTime: Long = {
    time
  }

  def setTime(time: Long): AudFrame = {
    this.time = time
    this
  }

  override def close(): Unit = {
    super.close()
    samples = null
  }
}
